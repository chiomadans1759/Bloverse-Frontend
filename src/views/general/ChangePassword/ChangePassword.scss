@import 'sass-rem';

#bl-change-password {
  height: 100vh;
  .main-content {
    padding: 0;
    .change-password-form {
      margin: 0;
      justify-content: center;
      .bl-form-header {
        margin-bottom: 2rem !important;
        &>h3 {
          color: #252525;
          font-weight: bold;
          text-transform: capitalize;
        }
      }
      .bl-form-btn {
        background-image: linear-gradient( to right, #003CE1, #2F80ED);
      }
      input, button {
        border-radius: 2px !important;
        font-size: rem(16px);
      }
      input::placeholder {
        color: #B7B7B7;
      }
    }
  }
  .footer-container {
    position: fixed;
    bottom: 0;
    color: #B7B7B7;
    line-height: rem(40px);
  }
}