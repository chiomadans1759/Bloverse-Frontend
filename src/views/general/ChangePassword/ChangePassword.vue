<template>
  <main id="bl-change-password" class="container-fluid p-0 bl-change-password">
    <navbar :showBackArrow="showBackArrow" :showNavigations="showNavigations" customClass="back-nav"/>
    <div class="container main-content">
      <div class="row change-password-form">
        <form>
          <div class="form-group bl-form-header">
            <h3>Change Password</h3>
          </div>
          <div class="form-group">
            <input type="password" class="form-control" id="old-password"
              aria-describedby="oldPassword" placeholder="Old Password"
            >
          </div>
          <div class="form-group">
            <input type="password" class="form-control" id="new-password"
              aria-describedby="newPassword" placeholder="New Password"
            >
          </div>
          <div class="form-group">
            <input type="password" class="form-control"
              id="confirm-password" placeholder="Confirm Password"
            >
          </div>
          <button type="submit" class="btn btn-primary btn-block bl-form-btn">
            Save
          </button>
        </form>
      </div>
    </div>
    <div class="footer-container">
      <p><span class="bloverse">bloverse</span> 2019. All Rights Reserved</p>
    </div>
  </main>
</template>

<script>
import Navbar from '@/components/Navbar/Navbar.vue';

export default {
  name: 'ChangePassword',
  components: { Navbar },
  data() {
    return {
      showNavigations: false,
      showBackArrow: true,
    }
  }
}
</script>

<style lang="scss" scoped>
 @import './ChangePassword.scss';
</style>

