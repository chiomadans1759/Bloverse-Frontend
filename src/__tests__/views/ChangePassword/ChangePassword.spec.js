import { mount, RouterLinkStub  } from '@vue/test-utils'
import ChangePassword from '@/views/consumers/ChangePassword/ChangePassword.vue';

describe('ChangePassword Component', () => {
  it('should render correctly', () => {

    const wrapper = mount(ChangePassword, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
