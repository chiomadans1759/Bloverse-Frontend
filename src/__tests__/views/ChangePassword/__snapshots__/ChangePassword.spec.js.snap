// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePassword Component should render correctly 1`] = `
<main
  class="container-fluid p-0 bl-change-password"
  id="bl-change-password"
>
  <header
    id="feeds-header"
  >
    <nav
      class="navbar navbar-expand-lg navbar-light px-5"
    >
      <div
        class="container-fluid back-nav"
      >
        <a
          class="backlink mr-4"
          href="/settings"
        >
          <div
            class="btn rounded-btn-circle d-flex justify-content-center"
          >
            <i
              class="fal fa-arrow-left back-icon"
            />
          </div>
        </a>
         
        <span
          class="back-text"
        >
          Back
        </span>
         
        <a
          class="router-link"
        >
          <img
            class="nav-logo"
            src="@/assets/Bloverse.svg"
          />
           
          <!---->
        </a>
         
        <!---->
         
        <!---->
      </div>
    </nav>
     
    <main
      id="bl-modal"
    >
      <div
        aria-hidden="true"
        aria-labelledby="customModal"
        class="modal fade"
        id="feedback-modal"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="modal-dialog modal-dialog-centered"
          role="document"
        >
          <div
            class="modal-content"
          >
            <div
              class="modal-header darkBackground"
            >
              <!---->
               
              <div>
                <h6>
                  <strong
                    class="alt-title"
                  >
                    Send Us Some Feedback
                  </strong>
                </h6>
                 
                <h6
                  class="mt-2 sub-title"
                >
                  Found a bug? Have a suggestion?. Fill out the form below and we'll take a look!
                </h6>
              </div>
               
              <button
                aria-label="Close"
                class="close"
                data-dismiss="modal"
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  Ã—
                </span>
              </button>
            </div>
             
            <div
              class="modal-body"
            >
              <div
                class="bl-modal-content"
              >
                <p>
                  Enter your feedback here!
                </p>
                 
                <textarea
                  class="form-control mb-2 p-1"
                />
                 
                <label
                  class="ml-1 mr-3"
                >
                  <input
                    name="feedback-type"
                    type="radio"
                  />
                    Bug
                </label>
                 
                <label
                  class="mr-3"
                >
                  <input
                    name="feedback-type"
                    type="radio"
                  />
                    Comment
                </label>
                 
                <label
                  class="mr-3"
                >
                  <input
                    name="feedback-type"
                    type="radio"
                  />
                    Other
                </label>
              </div>
            </div>
             
            <div
              class="modal-footer"
            >
              <button
                class="btn btn-outline-primary"
                data-dismiss="modal"
                type="button"
              >
                
              Close
            
              </button>
               
              <button
                class="btn btn-primary"
                type="button"
              >
                Send Feedback
              </button>
            </div>
          </div>
        </div>
      </div>
    </main>
  </header>
   
  <div
    class="container main-content"
  >
    <div
      class="row change-password-form"
    >
      <form>
        <div
          class="form-group bl-form-header"
        >
          <h3>
            Change Password
          </h3>
        </div>
         
        <div
          class="form-group"
        >
          <input
            aria-describedby="oldPassword"
            class="form-control"
            id="old-password"
            placeholder="Old Password"
            type="password"
          />
        </div>
         
        <div
          class="form-group"
        >
          <input
            aria-describedby="newPassword"
            class="form-control"
            id="new-password"
            placeholder="New Password"
            type="password"
          />
        </div>
         
        <div
          class="form-group"
        >
          <input
            class="form-control"
            id="confirm-password"
            placeholder="Confirm Password"
            type="password"
          />
        </div>
         
        <button
          class="btn btn-primary btn-block bl-form-btn"
          type="submit"
        >
          
          Save
        
        </button>
      </form>
    </div>
  </div>
   
  <div
    class="footer-container"
  >
    <p>
      <span
        class="bloverse"
      >
        bloverse
      </span>
       2019. All Rights Reserved
    </p>
  </div>
</main>
`;
